<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1201</width>
    <height>491</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1201</width>
    <height>491</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1201</width>
    <height>491</height>
   </size>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="acceptDrops">
    <bool>false</bool>
   </property>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>10</y>
      <width>791</width>
      <height>421</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Made</string>
     </attribute>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QTableWidget" name="tableWidget">
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>90</number>
        </attribute>
        <attribute name="horizontalHeaderMinimumSectionSize">
         <number>20</number>
        </attribute>
        <column>
         <property name="text">
          <string>Add?</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Transition Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Activity</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Station</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConArr</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConArr Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConWait</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConWait Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Excel row #</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Duplicates (will not add)</string>
     </attribute>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QTableWidget" name="tableWidget_2">
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>90</number>
        </attribute>
        <column>
         <property name="text">
          <string/>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Transition Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Activity</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Station</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConArr</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConArr Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConWait</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConWait Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Excel row #</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Failed</string>
     </attribute>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QTableWidget" name="tableWidget_3">
        <column>
         <property name="text">
          <string>Error code</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Station</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConArr</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConArr Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConWait</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ConWait Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Excel row #</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>381</width>
      <height>191</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>211</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Find connections with UD station name</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="stationnamebox">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>10</y>
       <width>133</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Edinburgh</string>
     </property>
     <property name="placeholderText">
      <string>e.g. 'Edinburgh'</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>40</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>that has TIPLOC code</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="tiplocbox">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>40</y>
       <width>101</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>EDINBUR</string>
     </property>
     <property name="placeholderText">
      <string>e.g. 'EDINBUR'</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_3">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>150</y>
       <width>161</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Find connections</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>between the times</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="mintimebox">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>70</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="placeholderText">
      <string>00:00:00</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="maxtimebox">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>70</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="placeholderText">
      <string>23:59:59</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="highlightbox">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>140</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="layoutDirection">
      <enum>Qt::RightToLeft</enum>
     </property>
     <property name="text">
      <string>Highlight Excel rows?</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="findallbox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>191</width>
       <height>21</height>
      </rect>
     </property>
     <property name="layoutDirection">
      <enum>Qt::RightToLeft</enum>
     </property>
     <property name="text">
      <string>Find all?</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="thresholdbox">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>100</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>10</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>19</x>
       <y>100</y>
       <width>241</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>and accept a maximum RSX-UD time difference of</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>100</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>minutes</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>381</width>
      <height>111</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>11</x>
       <y>11</y>
       <width>359</width>
       <height>91</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="3" column="1">
       <widget class="QLineEdit" name="lineEdit_2">
        <property name="text">
         <string>C:/Users/Tfarhy/OneDrive - Network Rail/2020.11.24_XML backend for adding connections/uAvanti.xlsx</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QLineEdit" name="lineEdit">
        <property name="text">
         <string>C:/Users/Tfarhy/OneDrive - Network Rail/2020.11.24_XML backend for adding connections/donovan10.rsx</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QComboBox" name="udselector"/>
      </item>
      <item row="3" column="3">
       <widget class="QPushButton" name="pushButton_2">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Unit Diagram:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>RSX:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Loc. mapping</string>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QPushButton" name="pushButton_4">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="2">
       <widget class="QLineEdit" name="lineEdit_3">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="saveButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>440</y>
      <width>221</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Add connections to RSX and save as...</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="console">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>381</width>
      <height>131</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>120</red>
          <green>120</green>
          <blue>120</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>240</red>
          <green>240</green>
          <blue>240</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="frameShape">
     <enum>QFrame::VLine</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="debugbutton">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>440</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Debug</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>rsxbrowse_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>552</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>udbrowse_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>587</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>generate_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>587</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>savebutton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1070</x>
     <y>451</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>debugbutton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>debugbutton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>437</x>
     <y>451</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableWidget</sender>
   <signal>itemChanged(QTableWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>cellChangedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>794</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>locationmappingbrowse_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>939</x>
     <y>421</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>rsxbrowse_clicked()</slot>
  <slot>udbrowse_clicked()</slot>
  <slot>generate_clicked()</slot>
  <slot>savebutton_clicked()</slot>
  <slot>debugbutton_clicked()</slot>
  <slot>cellChangedSlot()</slot>
  <slot>cellActivatedSlot()</slot>
  <slot>locationmappingbrowse_clicked()</slot>
 </slots>
</ui>
